Library/Homebrew/extend/os/linux/dev-cmd/determine-test-runners.rb

```diff
@@ -0,0 +1,197 @@
+# typed: false
+# frozen_string_literal: true
+
+require "formula"
+
+class Formula
+  def macos_only?
+    requirements.any? { |r| r.is_a?(MacOSRequirement) && !r.version_specified? }
+  end
+
+  def linux_only?
+    requirements.any?(LinuxRequirement)
+  end
+
+  def x86_64_only?
+    requirements.any? { |r| r.is_a?(ArchRequirement) && (r.arch == :x86_64) }
+  end
+
+  def arm64_only?
+    requirements.any? { |r| r.is_a?(ArchRequirement) && (r.arch == :arm64) }
+  end
+
+  def versioned_macos_requirement
+    requirements.find { |r| r.is_a?(MacOSRequirement) && r.version_specified? }
+  end
+
+  def compatible_with?(macos_version)
+    return true if versioned_macos_requirement.blank?
+
+    macos_version.public_send(versioned_macos_requirement.comparator, versioned_macos_requirement.version)
+  end
+
+  def dependents
+    @dependent_hash ||= {}
+    @dependent_hash[ENV["HOMEBREW_SIMULATE_MACOS_ON_LINUX"].present?] ||= with_env(HOMEBREW_STDERR: "1") do
+      Utils.safe_popen_read(
+        HOMEBREW_BREW_FILE, "uses", "--formulae", "--eval-all", "--include-build", "--include-test", name
+      ).split("\n").map { |dependent| Formula[dependent] }.freeze
+    end
+
+    @dependent_hash[ENV["HOMEBREW_SIMULATE_MACOS_ON_LINUX"].present?]
+  end
+end
+
+module Homebrew
```

Comment:
Similarly having this all live in another `module` than `Homebrew` would be nice.

Reply:
Will need to look at how `CLI::Parser` works, since all other commands use `module Homebrew`.

Reply:
I think the only ways to do this require one of the following:

- the separate `module` will need to be mixed into `Homebrew` (which kind of misses the point)
- Our handling of new-style Ruby commands will need to be modified to work with commands that live in a separate module, since we currently invoke these commands with `Homebrew.send` (I don't really have a clean implementation in mind for this yet.)
- Write this command as an old-style Ruby command that only needs to be `require`d (but then I think we don't get to use `Homebrew::CLI::Parser`)

Not really all that keen on any of the above options, but let me know if I'm missing something.

Reply:
`def self.determine_test_runners` needs to and can remain in `Homebrew`. Most of the code inside it, though, could be moved to any other module and called in there e.g. 

```ruby
	 sig { void }
   def self.determine_test_runners
     args = determine_test_runners_args.parse

     Homebrew::DetermineTestRunners.do_stuff(args)
   end
```

This will also make it easier to do unit testing and not have to have all this logic live in a Linux-only file.